akka-http/src/main/scala/akka/http/scaladsl/server/directives/CodingDirectives.scala

```diff
@@ -107,6 +108,16 @@ trait CodingDirectives {
    */
   def decodeRequest: Directive0 =
     decodeRequestWith(DefaultCoders: _*)
+
+  /**
+   * Inspects the response entity and adds a `Content-Encoding: gzip` response header if
+   * the entities media-type is precompressed with gzip and no `Content-Encoding` header is present yet.
```

Comment:
Ok. I once experimented with an impl that for nested coding directives peeled off `Content-Encoding` values while decoding, and stacked them for encoding: https://github.com/2beaucoup/akka/commit/b32aeb5ebcc4b433bc8803b0dbcd921da078651b

The current impl just replaces the whole header so this looks ok.

You're off the hook now. ;)